modules/data.land/R/IC_BADM_Utilities.R

```diff
@@ -0,0 +1,296 @@
+          
+        } else if (type == "*SOIL") {
+          val <- Gdf %>%
+            dplyr::filter(VARIABLE %>% grepl("SOIL_STOCK_C_ORG", .)) %>% #"SOIL_STOCK_C_ORG"
+            dplyr::pull(DATAVALUE) %>%
+            as.numeric()
+          
+          if (length(val) > 0)
+            SoilIni <- udunits2::ud.convert(val,  "g/m^2", "kg/m^2")
+          
+        } else if (type == "*_LIT_BIOMASS") {
+          litterIni <-
+            udunits2::ud.convert(Gdf$DATAVALUE[1],  unit.ready, "kg/m^2")
+          
+        } else if (type == "*_ROOT_BIOMASS") {
+          Rootini <-
+            udunits2::ud.convert(Gdf$DATAVALUE[1],  unit.ready, "kg/m^2")
+          
+        }
+        return(
+          data.frame(
+            Site = Gdf$SITE_ID %>% unique(),
+            Var = Gdf$VARIABLE[1],
+            Date = Date.in,
+            Organ = Oregan.in,
+            PlantWIni = PlantWoodIni,
+            SoilIni = SoilIni,
+            LitterIni = litterIni,
+            RootIni = Rootini
+          )
+        )
+    })
+ #cleaning
+ind <- apply(entries[,5:8], 1, function(x) all(is.na(x)))
+entries <- entries[-which(ind),]
+
+
+
+  return(entries)
+}
+
+
+#' netcdf.writer.BADAM
+#'
+#' @param lat numeric latitude
+#' @param long numeric longitude
+#' @param siteid site id as a string
+#' @param outdir outputdir which you want to store the IC netcdf file
+#'
+#' @return a string of the address where the IC file is stored
+#' @export
+#'
+netcdf.writer.BADAM <- function(lat, long, siteid, outdir ){
+  #--
+  input <- list()
+  dims <- list(lat = lat ,
+               lon = long,
+               time = 1)
+
+  #Reading in the BADM data
+  entries <- Read.IC.info.BADM (lat, long)
+  
+  PWI <- entries$PlantWIni[!is.na(entries$PlantWIni)]
+  LIn <- entries$LitterIni[!is.na(entries$LitterIni)]
+  SIn <- entries$SoilIni[!is.na(entries$SoilIni)]
+  
+  
+  variables <- list(
+    SoilMoistFrac = 1,
+    SWE = 0
+  )
+  
+  if (length(PWI)>0 ) variables <- c(variables, wood_carbon_content = sample(PWI, 1, replace = T))
+  if (length(LIn)>0 ) variables <- c(variables, litter_carbon_content = sample(LIn, 1, replace = T))
+  if (length(SIn)>0 ) variables <- c(variables, soil_organic_carbon_content = sample(SIn, 1, replace = T))
+  
+  input$dims <- dims
+  input$vals <- variables
+  
+  return(pool_ic_list2netcdf(
+    input = input,
+    outdir = outdir,
+    siteid = siteid
+  )$file)
+}
+
+
+#' IC_Maker
+#'
+#' @param settings PEcAn xml settings
+#' @param ens.n number of ensemble memebers needs to be genrated
+#' @param outdir output directory 
+#' @description This function generates a series of ensemble members for initial condition using the BADM database.
+#' @return xml setting
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' settings <- PEcAn.settings::read.settings("pecan.SDA.10sites.xml")
+#' 
+#'  suppressMessages({
+#'   setting.with.ic <- PEcAn.settings::papply(settings,
```

Comment:
Given structure of ic_process (and exactly where you add this to the existing workflow), it probably makes more sense to just return a convert.inputs results block, input IDs, or settings$run$inputs, and not to modify/return the entire settings object

Reply:
`netcdf.writer.BADAM` function is designed to do this. So in the example if we put `netcdf.writer.BADAM` in the papply it would return just a path to a file.

Reply:
Ok, but that then begs the question as to why this function exists if the workflow is supposed to call netcdf.writer.BADM directly? As far as I can tell this function is unnecessary, can be dropped, and netcdf.writer.BADM should be updated to return a results dataframe (file, host, start_date, end_date, mimetype, formatname) instead of just the path

Will also note that this function name is very non-specific (which may not matter if it's removed), and that you should fix the typo in netcdf.writer.BADM function name.